source/develop/release-management/features/storage/incremental-backup.html.md

```diff
@@ -12,173 +12,421 @@ feature_status: Design
+        <disk id="disk-uuid" />
+        ...
+    </disks>
+</backup>
+```
+
+Response:
+```
+<backup id="backup-uuid">
+    <incremental>previous-backup-uuid</incremental>
+    <disks>
+        <disk id="disk-uuid" />
+        ...
+        ...
+    </disks>
+    <status>initiailizing</status>
+    <creation_date>...
+</backup>
+```
+
+#### Getting backup info
+
+When backup status is "ready", you can start downloading the disks.
+
+Request:
+```
+GET /vms/vm-uuid/backups/backup-uuid
+```
+
+Response:
+```
+<backup id="backup-uuid">
+    <incremental>previous-backup-uuid</incremental>
+    <disks>
+        <disk id="disk-uuid">
+            <image_id>image-uuid</image_id>
+        </disk>
+        ...
+    </disks>
+    <status>ready</status>
+    <creation_date>...
+</backup>
+```
+
+#### Finalizing backup
+
+```
+POST /vms/vm-uuid/backups/backup-uuid/finalize
+
+<action></action>
+```
+
+#### Canceling backup
+
+```
+POST /vms/vm-uuid/backups/backup-uuid/cancel
+
+<action></action>
+```
+
+#### Creating image transfer for incremental restore
+
+XXX Write me
+
+### Incremental backup ticket example
+
+For running VM, qemu will serve the disk using NBD:
+```
+{
+    "url": "nbd:localhost:1234:exportname=/sda"
+}
+```
+
+For stopped VM, we will run one qemu-nbd instance per disk, using unix
+socket:
+```
+{
+    "url": "nbd:unix:/run/vdsm/nbd/xxxyyy-nbd.sock:exportname=/sda"
+}
+```
+
+### Open Issues
+
+- How snapshots and checkpoints interract? The most useful way seems to
+  be closing the current bitmap on the old active layer, and starting a
+  new bitmap on the new active layer.
+
+- Handle the case when previous-backup-uuid does not exist on one of the
+  disks. Possible use cases:
+  - attaching a new disk to a VM.
+  - checkpoint was deleted from one of the disks.
+
+- Handle the case when all disks have previous-backup-uuid, but
+  different checkpoints in the middle.
+
+- Handle the case when one checkpoint is missing in a chain. Backing up
+  the existing checkpoints will miss the data referred by the missing
+  snapshot.
+
+- Handle VM stop during a backup - should abort the backup, causing
+  image transfers to fail.
+
+- Handle VM start during a backup of an offline VM - should abort the
+  backup, causing image transfers to fail.
+
+- Locking disks during backup - currently image transfer lock the disk,
+  but we need to lock the disk when creating the backup. Image transfer
+  should skip locking if the disk is included in incremental backup.
+
+- How to handle failure on finalize and cancel? For example failure
+  to delete a scratch disk. Should the system continue to track the
+  items and retry to complete the operation later?
+
+
+## Links
+
+- RFC v2 external (pull) backup API:
+
+    https://www.redhat.com/archives/libvir-list/2018-April/msg00115.html
+
+- VMWare CBT docs:
+
+    https://kb.vmware.com/s/article/1020128
+
+- Eric Blake patches:
+
+    https://www.redhat.com/archives/libvir-list/2018-June/msg01066.html
```

Comment:
Add link to Eric talk in KVM Forum 2018:
https://events.linuxfoundation.org/wp-content/uploads/2017/12/Eric-Blake_2018-libvirt-incremental-backup.pdf